cmds/legacy_boot/legacy_boot.go

```diff
@@ -0,0 +1,383 @@
+// Copyright 2012-2017 the u-root Authors. All rights reserved
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+// legacy_boot allows to handover a system running linuxboot/u-root
+// to a legacy preinstalled operating system by replacing the traditional
+// bootloader path
+
+//
+// Synopsis:
+//	legacy_boot
+//
+// Description:
+//	If returns to u-root shell, the code didn't found a local bootable option
+//
+// Notes:
+//	The code is looking for boot/grub/grub.cfg file as to identify the
+//	boot option.
+//	The first bootable device found in the block device tree is the one used
+//	Windows is not supported (that is a work in progress)
+//
+// Example:
+//	legacy_boot -v 	- Start the script in verbose mode for debugging purpose
+
+package main
+
+import (
+	"bufio"
+	"fmt"
+	"github.com/u-root/u-root/pkg/kexec"
+	"io"
+	"io/ioutil"
+	"log"
+	"os"
+	"strings"
+	"flag"
+	"syscall"
+)
+
+var verbose bool
+type options struct {
+        verbose bool
+}
+
+func check(e error) {
```

Comment:
I generally try to follow the advice found at https://golang.org/doc/effective_go.html.

One of the ones I've started being a pain about for this project is https://golang.org/doc/effective_go.html#panic - tl;dr: don't panic unless it's unrecoverable.

I can imagine wanting to use some of these functions in other scripts via a package, and if we are going to eventually packetize this, we should use normal Go error reporting instead of panicking.